src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp

```diff
@@ -1823,6 +1835,21 @@ FixedwingPositionControl::tecs_update_pitch_throttle(float alt_sp, float airspee
 					     accel_body, (_global_pos.timestamp > 0), in_air_alt_control,
 					     _global_pos.alt, _local_pos.v_z_valid, _local_pos.vz, _local_pos.az);
 
+	/* scale throttle cruise by baro pressure */
+	if (_parameters.throttle_alt_scale) {
+
+		bool baro_updated = false;
+		orb_check(_sensor_baro_sub, &baro_updated);
+
+		sensor_baro_s baro;
+
+		if (orb_copy(ORB_ID(sensor_baro), _sensor_baro_sub, &baro) == PX4_OK) {
+			// scale throttle as a function of sqrt(p0/p) (~ EAS -> TAS at low speeds and altitudes ignoring temperature)
+			const float scale = constrain(sqrtf(MSL_PRESSURE_MILLIBAR / baro.pressure), 0.9f, 1.1f);
+			throttle_cruise = throttle_cruise * scale;
```

Comment:
throttle_max should also be scaled up to a max value of 1.0 to prevent possible loss of speed an stalling during climbs at altitude.

Reply:
Can you elaborate on 1.0? shouldnt the same scaling apply? or do you mean as a lower limit?

Reply:
Or do you mean an absolute value of 1.0f ?